internal/buildscripts/packaging/fpm/test.sh

```diff
@@ -30,47 +30,46 @@ if [[ ! "$pkg_type" =~ ^(deb|rpm)$ ]]; then
 fi
 image_name="otelcontribcol-$pkg_type-test"
 container_name="$image_name"
-docker_run="docker run --name $container_name -d -v /sys/fs/cgroup:/sys/fs/cgroup:ro --privileged $image_name"
-docker_exec="docker exec $container_name"
+container_exec="podman exec $container_name"
 
-trap "docker rm -fv $container_name >/dev/null 2>&1 || true" EXIT
+trap "podman rm -fv $container_name >/dev/null 2>&1 || true" EXIT
 
-docker build -t $image_name -f "$SCRIPT_DIR/$pkg_type/Dockerfile.test" "$SCRIPT_DIR"
-docker rm -fv $container_name >/dev/null 2>&1 || true
+podman build -t $image_name -f "$SCRIPT_DIR/$pkg_type/Dockerfile.test" "$SCRIPT_DIR"
+podman rm -fv $container_name >/dev/null 2>&1 || true
 
 # test install
 echo
-$docker_run
+podman run --name $container_name -d $image_name
```

Comment:
In your comment [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16450#issuecomment-1867077311), you said there should be a check to ensure that the `run` does not silently fail. Will podman fail here if the container fails to start or should we still add an additional check?

Reply:
Yes, `podman-run` will return a non-zero exit code ala `chroot`, 125 - podman command itself failed, 126 - the contained command failed to run, 127 - the contained command could not be found. (These are the same as `docker-run` should return)

However, the original issue in #16450 is actually the result of the initial `docker-run` to succeeding, but then the container immediately shuts off again due to unknown reasons (apart from the reason that Docker and SystemD have been at loggerheads for years and expecting the two to run smoothly together is not an option at this point in time)

So, going back to my original comment in your question. Yes,  there should be a health-check command running inside the `otelcolcontrib-*-test` images that waits until SystemD has finished starting up all services before returning a 'healthy' state. Then the `test.sh` bash script could wait for the container to transition into a healthy state after `podman-run` and before `podman-cp`, which would cover for that period of uncertainty where the container started, but might have not fully initialized as expected.

I think this should be part of a separate patch though. I'll start working on adding that health check function.

Reply:
Thanks for the thorough answer. If this PR gets merged before the health check function PR is submitted please create a new tracking issue for it.

Reply:
I created a tracking issue for the health check issue. #30239